### PR TITLE
New Resource: `azurerm_virtual_hub_routing_intent`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/21701

--- PASS: TestAccVirtualHubRoutingIntent_basic (3342.91s)
--- PASS: TestAccVirtualHubRoutingIntent_requiresImport (3478.10s)
--- PASS: TestAccVirtualHubRoutingIntent_update (3612.53s)

### CODE REVIEW

File: website/docs/r/virtual_hub_routing_intent.html.markdown [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Virtual Hub Routing Intent. Changing this forces a new resource to be created.
+
+* `virtual_hub_id` - (Required) The resource ID of the Virtual Hub. Changing this forces a new resource to be created.
+
+* `routing_policy` - (Required) A `routing_policy` block as defined below.
```
Comment (catriona-m) [inline website/docs/r/virtual_hub_routing_intent.html.markdown:68]:
```suggestion
* `routing_policy` - (Required) One or more `routing_policy` blocks as defined below.
```

File: internal/services/network/virtual_hub_routing_intent_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
```
Comment (catriona-m) [inline internal/services/network/virtual_hub_routing_intent_resource_test.go:66]: can we add another test step to change the values within the `routing_policy`s to confirm that updating the blocks works as expected?

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks for this @neil-yechenwei - I left a couple of minor comments for you to consider, but can take another look once those are addressed. Thanks!

File: website/docs/r/virtual_hub_routing_intent.html.markdown (hunk 1 above)
Comment (neil-yechenwei) [reply website/docs/r/virtual_hub_routing_intent.html.markdown]: Updated

File: internal/services/network/virtual_hub_routing_intent_resource_test.go (hunk 2 above)
Comment (neil-yechenwei) [reply internal/services/network/virtual_hub_routing_intent_resource_test.go:73]: Updated

**Comment by neil-yechenwei** [comment]
@catriona-m , thanks for the comment. I updated PR. Please take another look. Thanks.

image

**Review by catriona-m — APPROVED** [review]
Thanks for making those changes @neil-yechenwei - LGTM 🎺 !

**Pull request merged**